public/contextMenu.js

```diff
@@ -0,0 +1,258 @@
+/*
+MIT License
+
+Copyright (c) Sindre Sorhus <sindresorhus@gmail.com> (https://sindresorhus.com)
+Original from: https://github.com/sindresorhus/electron-context-menu/releases/tag/v3.1.1
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documen
+    tation files (the "Software"), to deal in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+    persons to whom the Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+    Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+    WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+    OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+    OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+const electron = require('electron');
+
+const webContents = win => win.webContents || (win.id && win);
+
+const decorateMenuItem = menuItem => {
+  return (options = {}) => {
+    if (options.transform && !options.click) {
+      menuItem.transform = options.transform;
+    }
+
+    return menuItem;
+  };
+};
+
+const removeUnusedMenuItems = menuTemplate => {
+  let notDeletedPreviousElement;
+
+  return menuTemplate
+    .filter(menuItem => menuItem !== undefined && menuItem !== false && menuItem.visible !== false && menuItem.visible !== '')
+    .filter((menuItem, index, array) => {
+      const toDelete = menuItem.type === 'separator' && (!notDeletedPreviousElement || index === array.length - 1 || array[index + 1].type === 'separator');
+      notDeletedPreviousElement = toDelete ? notDeletedPreviousElement : menuItem;
+      return !toDelete;
+    });
+};
+
+const create = (win, options) => {
+  const handleContextMenu = (event, props) => {
+    if (typeof options.shouldShowMenu === 'function' && options.shouldShowMenu(event, props) === false) {
+      return;
+    }
+
+    const {editFlags} = props;
+    const hasText = props.selectionText.trim().length > 0;
+    const can = type => editFlags[`can${type}`] && hasText;
+
+    const defaultActions = {
+      separator: () => ({type: 'separator'}),
+      cut: decorateMenuItem({
+        id: 'cut',
+        label: 'Cu&t',
```

Comment:
What's this &?

Reply:
& is a syntax for the context menu to know where to put the underscore on the word. And then, what "hotkey" will be assigned to that button.

In that case, a right click shows the context menu and if the user types `T` on the keyboard, the system runs the "Cut" operation.